### PR TITLE
Parameterize bucket and pattern used to generate filename (_prefix)

### PR DESCRIPTION
**Pull request by f-skold**
I want to reuse this code, but use my own S3 bucket in the same region that our website runs in.

Some reformatting was done by the commit hooks when I did the work.

### CODE REVIEW

**Review by kylebarron — APPROVED** [review]

File: rio_tiler_pds/sentinel/utils.py
Comment (vincentsarago) [reply rio_tiler_pds/sentinel/utils.py:32]: I think the diff comes from black

Comment (f-skold) [reply rio_tiler_pds/sentinel/utils.py:32]: Don't know why, but some tool did reformat that automatically.
Could it be that I did upgrade black 21.10b0 -> 22.3.0, and that tool seemed to start do its work when I touched a file.
I got import error from black, and this post https://stackoverflow.com/questions/71673404/importerror-cannot-import-name-unicodefun-from-click suggested the update to 22.3.0.

So I guess that if I try to revert it, the automatic tool will reapply the change.

But this is the first time I do this, so I can be wrong.

**Comment by f-skold** [comment]
I don't know how this normally works. I the approval from @kylebarron enough, so I should do something? In that case, what? 
Or am I waiting for others to approve it?

**Comment by kylebarron** [comment]
@vincentsarago has the final say on any changes 🙂

File: rio_tiler_pds/cbers/aws/cbers4.py
Comment (vincentsarago) [inline rio_tiler_pds/cbers/aws/cbers4.py:37]: What do you think about adding the `scheme` as a parameter too?

Comment (f-skold) [reply rio_tiler_pds/cbers/aws/cbers4.py:37]: Your the boss :-)
If you think that it is a good idea, or if you want everything to be customizable, I can update it like that...

I think I did have that thought a short while, but thought that if it is in AWS directory, and using different ways of retriving the metadata-json-file (more of a pure boto s3 call). It did not make that much sense to enable other schemes.

And if one were to use "ftp" or "http", the term "hostname" -> "bucket" would be misleading in that case. (But mayby the main case where scheme is s3, and bucket is the more appropriate name for that parameter.

Actually, isn't the "reuquester pays" buckets public?
I did use environment variables to insert the region name into the hostname rasterio constucts, and that was in what got it working for me.

Comment (vincentsarago) [reply rio_tiler_pds/cbers/aws/cbers4.py:37]: oh I forgot about the `get_object`, then it doesn't make sense. The Reader should only work with S3 prefix

Comment (f-skold) [reply rio_tiler_pds/cbers/aws/cbers4.py:37]: Now I checked the sentinel2.py, the other class S2L2AReader does not read the json file during construction of  the object.

This might complicate things, but one could of course later do something like this:
```
    if "s3" == self.scheme:
        json_body = get_object(self.bucket, f"{prefix}/tileInfo.json", request_pays=True)
    else:
        # Do stuff. Looking at rio_tiler, it might be like:
        src_path = ""
        with rasterio.Env(**self.gdal_config):
                jsonbody = self.reader(src_path, **reader_params):
    self.tileInfo = json.loads(jsonbody)
```
If we take this step, we might take another and inject the reader and have a single way of retirering the data.

Comment (f-skold) [reply rio_tiler_pds/cbers/aws/cbers4.py:37]: @vincentsarago : Is this pull request "good enough" so it could be merged? Or should I do anything more? 

Should I redo the update of black reformating all python code with the new version? (In the current version, some things ended up in somewhat strange places.

File: rio_tiler_pds/sentinel/utils.py
Comment (f-skold) [reply rio_tiler_pds/sentinel/utils.py:32]: I did revisit the "black" upgrade. And this think did not show up from black.
So I suppose it must have been me :-) Removing it, and I might think about doing that change for the entire code base.

https://docs.python.org/3/library/re.html?highlight=re#module-re

> The solution is to use Python’s raw string notation for regular expression patterns; backslashes are not handled in any special way in a string literal prefixed with 'r'. So r"\n" is a two-character string containing '\' and 'n', while "\n" is a one-character string containing a newline. Usually patterns will be expressed in Python code using this raw string notation.

File: rio_tiler_pds/cbers/aws/cbers4.py
Comment (vincentsarago) [reply rio_tiler_pds/cbers/aws/cbers4.py:37]: sorry I'll take care of this today!

**Pull request merged**
